### DISCUSSION TITLE
min peptide length not working v1.8

### DISCUSSION THREAD

**Discussion by alexproteomics** [vdemichev/DiaNN#726] [General]
Hi Vadim,

I'm particularly interested in identifying a 6-aminoacid peptide from a certain protein. My data is diaPASEF and I searched in library-free mode (BTW, still on version 1.8.0) after setting the min-pep-len option to 6. I can see see it on the command output, but my report file does not have any 6-aa peptide. I also opened the report-lib.tsv and the report-lib.prosit.csv and could not find any 6-aa peptide. 
Am I doing something wrong?
Here is my command:
diann.exe --f ALL-MY-RAW-FILES  --lib  --threads 28 --verbose 1 --out C:\Projects\AMG2\mtPIK3CA\report.tsv --qvalue 0.01 --matrices --out-lib C:\Projects\TPD302\report-lib.tsv --gen-spec-lib --predictor --fasta C:\Projects\db\uniprot-reviewed_HUMAN.fasta --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --met-excision --cut K*,R* --missed-cleavages 1 --min-pep-len 6 --max-pep-len 30 --min-pr-mz 250 --max-pr-mz 1800 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --mass-acc 15 --mass-acc-ms1 15 --individual-mass-acc --individual-windows --smart-profiling --pg-level 0 --peak-center --no-ifs-removal --no-norm --report-lib-info 

BTW, I also tried to generate the library alone (no search files) and also no luck in getting any 6-aa peptide.

Thanks
Alex

**Comment by alexproteomics** [comment]
update: I downloaded version 1.8.1 and still have the same problem, the shortest peptides in the results are 7 aa long even though I set min-pep-len to 6

**Reply by vdemichev** [reply]
Hi Alex,

I would think this is due to filters applied to (i) minimum fragment length when generating lib from FASTA, which is 3 amino acids, (ii) minimum fragment mass detectable in the experiment / generated by DIA-NN when producing the library and (iii) minimum number of fragments allowed for the peptide to be considered (4 or greater). A peptide of 6 amino acids can have max 4 y/b series charge 1 fragments (charge 2 will not satisfy the min m/z bound).

Solution: generate the library with Prosit or some other deep learning tool, make sure it includes size 2 fragment ions (don't incude any size 1), use this lib with DIA-NN - shoudl work.

Best,
Vadim

**Reply by alexproteomics** [reply]
thanks for the tip Vadim. Just for clarification, do you mean to use DIANN to generate the predicted Prosit library with --min-fr ? If that is not the suggestion, I would appreciate further detail on how to generate the prosit speclib from a fasta file (I understand that the online Prosit only works for Thermo?!).
thanks

**Reply by vdemichev** [reply]
No, meant using Prosit or any other tool for this without DIA-NN being used to digest the FASTA.

**Reply by vdemichev** [reply]
FASTA digest for Prosit specifically can be prepared also with EncyclopeDIA
